js-api-spec/value/list.test.ts

```diff
@@ -0,0 +1,283 @@
+// Copyright 2021 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {Value, SassList, SassMap, SassNumber, SassString} from 'sass';
+import {List} from 'immutable';
+
+import {skipForImpl} from '../utils';
+
+skipForImpl('dart-sass', () => {
+  describe('SassList', () => {
+    describe('construction', () => {
+      const list = new SassList(
+        [new SassString('a'), new SassString('b'), new SassString('c')],
+        {separator: ','}
+      );
+
+      it('is a value', () => {
+        expect(list).toBeInstanceOf(Value);
+      });
+
+      it('is a list', () => {
+        expect(list).toBeInstanceOf(SassList);
+      });
+
+      it("isn't any other type", () => {
+        expect(() => list.assertBoolean()).toThrow();
+        expect(() => list.assertColor()).toThrow();
+        expect(() => list.assertFunction()).toThrow();
+        expect(() => list.assertMap()).toThrow();
+        expect(list.tryMap()).toBe(null);
+        expect(() => list.assertNumber()).toThrow();
+        expect(() => list.assertString()).toThrow();
+      });
+
+      it('returns its contents as a list', () => {
+        expect(
+          list.asList.equals(
+            List([
+              new SassString('a'),
+              new SassString('b'),
+              new SassString('c'),
+            ])
+          )
+        ).toBe(true);
+      });
+    });
+
+    describe('equality', () => {
+      const list = new SassList(
+        [new SassString('a'), new SassString('b'), new SassString('c')],
+        {separator: ','}
+      );
+
+      it('equals the same list', () => {
+        expect(list).toEqualWithHash(
+          new SassList(
+            [new SassString('a'), new SassString('b'), new SassString('c')],
+            {separator: ','}
+          )
+        );
+      });
+
+      it("doesn't equal a value with different metadata", () => {
+        expect(
+          list.equals(
+            new SassList(
+              [
+                new SassString('a', {quotes: false}),
+                new SassString('b', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ' '}
+            )
+          )
+        ).toBe(false);
+
+        expect(
+          list.equals(
+            new SassList(
+              [
+                new SassString('a', {quotes: false}),
+                new SassString('b', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ',', brackets: true}
+            )
+          )
+        ).toBe(false);
+      });
+
+      it("doesn't equal a value with different contents", () => {
```

Comment:
Also test equality with a shorter list and a longer list that have the same contents where they overlap.